src/workflows/airqo_etl_utils/data_api.py

```
@@ -923,12 +1094,41 @@
                 logger.exception("Method not supported")
                 return None
 
-            if response.status == 200 or response.status == 201:
+            return self.parse_api_response(response, url)
+
+        except urllib3.exceptions.HTTPError as http_err:
+            logger.exception(f"Client error: {http_err}")
+        except urllib3.exceptions.TimeoutError:
+            logger.exception(f"Timeout occurred: {http_err}")
```

Ruff (0.8.2)

Undefined name `http_err`

(F821)